### PR TITLE
windows: use FLUENT_PACKAGE_TOPDIR to work correctly

### PR DESCRIPTION
**Pull request by kenhys**
Use pre-defined variable to detect appropriate installed environment.

### CODE REVIEW

**Review by daipom — COMMENTED** [review]
Thanks for this fix!
I want to make it clear what this PR fixes.
 
This PR supports the case where the package is installed in the custom path on Windows,
and does not fix https://github.com/fluent/fluent-package-builder/issues/605, right?

File: lib/fluent/diagtool/diagutils.rb [hunk 1]
```diff
@@ -363,7 +363,13 @@ def diaglogger_error(str)
...
     def fluent_package?
-      File.exist?("/etc/fluent/fluentd.conf") || File.exist?("/opt/fluent/bin/fluentd")
+      if Diagtool.windows?
+        # FLUENT_PACKAGE_TOPDIR indicates installed directory, which is defined by fluent-package
+        File.exist?(File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "etc/fluent/fluentd.conf")) ||
+          File.exist?(File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "bin/fluentd"))
+      else
+        File.exist?("/etc/fluent/fluentd.conf") || File.exist?("/opt/fluent/bin/fluentd")
+      end
     end
```
Comment (daipom) [inline lib/fluent/diagtool/diagutils.rb]: This class should not have platform-specific logic.
How about fixing `lib/fluent/diagtool/windows/diagutils.rb`.

I believe that just adding the following function to the file is enough.

```rb
      def fluent_package?
        ENV.key?("FLUENT_PACKAGE_TOPDIR")
      end
```

File: lib/fluent/diagtool/collectutils.rb [hunk 2]
```diff
@@ -422,9 +422,17 @@ def match_platform?(platforms_option)
...
+                         "/opt/fluent/share/Gemfile"
+                       end
                      else
-                       "/opt/td-agent/share/Gemfile"
+                       if Diagtool.windows?
+                         File.join(ENV["TD_AGENT_TOPDIR"], "share/Gemfile")
+                       else
+                         "/opt/td-agent/share/Gemfile"
+                       end
                      end
```
Comment (daipom) [inline lib/fluent/diagtool/collectutils.rb]: This class should not have platform-specific logic.
How about fixing `lib/fluent/diagtool/windows/collectutils.rb`.

For example, we can separate the `gemfile_path` logic to a function:

`lib/fluent/diagtool/collectutils.rb`

```rb
    def collect_bundled_plugins
      File.read(gemfile_path).lines(chomp: true).grep(/\Agem "fluent-plugin-/).collect do |line|
        gem_name = line[/\Agem "(fluent-plugin-.+?)"/, 1]
        platforms_option = line[/platforms: (\w+)/, 1]
        next gem_name, platforms_option
      end.select do |gem_name, platforms_option|
        platforms_option.nil? || match_platform?(platforms_option)
      end.collect do |gem_name, platforms_option|
        gem_name
      end
    end

    (...)

    def gemfile_path
      if fluent_package?
        "/opt/fluent/share/Gemfile"
      else
        "/opt/td-agent/share/Gemfile"
      end
    end
```

And, we can add Windows-specific logic to `lib/fluent/diagtool/windows/collectutils.rb`

```rb
      def gemfile_path
        if fluent_package?
          File.join(ENV["FLUENT_PACKAGE_TOPDIR"], "share/Gemfile")
        else
          File.join(ENV["TD_AGENT_TOPDIR"], "share/Gemfile")
        end
      end
```

File: lib/fluent/diagtool/diagutils.rb (hunk 1 above)
Comment (kenhys) [reply lib/fluent/diagtool/diagutils.rb]: Thanks. It's appropriate fix.

File: lib/fluent/diagtool/collectutils.rb (hunk 2 above)
Comment (kenhys) [reply lib/fluent/diagtool/collectutils.rb]: Thanks!

**Comment by kenhys** [comment]
> and does not fix https://github.com/fluent/fluent-package-builder/issues/605, right?

I believe so.

**Review by daipom — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**